src/SelectEventPoll.cc

```diff
@@ -240,13 +240,16 @@ void checkFdCountMingw(const fd_set& fdset)
 
 void SelectEventPoll::updateFdSet()
 {
+  FD_ZERO(&rfdset_);
+  FD_ZERO(&wfdset_);
 #ifdef __MINGW32__
+  FD_SET(dummySocket_, &rfdset_);
+  FD_SET(dummySocket_, &wfdset_);
   fdmax_ = dummySocket_;
-#else  // !__MINGW32__
+#else  // __MINGW32__
```

Comment:
As far as I know, the convention in aria2 source code is negate condition on "#else".  So the original code is right.
It should be `// !__MINGW32__`